### PR TITLE
RMMIS-6177: FileList API GET support

### PR DESCRIPTION
**Pull request by kjschulz**
RMMIS-6177: Added support for calling a FileList API form the File component if no files are supplied as a value, added support for multiple File components on a page, fixed tests, wrote new tests for new actions, syntax fixes

### CODE REVIEW

File: src/File.jsx [hunk 1]
```diff
@@ -24,23 +31,36 @@ class File extends React.Component {
...
   }
 
   componentDidMount() {
+    // on fetch of file preview list on intial render
+    Dispatcher.register(`${this.props.id}-load-file-preview-list`, function(action, data) {
+      if (action === FILE_PREVIEW_LIST_LOAD && data.fieldId === this.props.id) {
+        // remember not to set value here, only files (will get security errors from file reader api)
+        this.setState({files: Array.isArray(data.value) ? data.value : []});
+      }
+    }.bind(this));
```
Comment (mallenscdhhs) [inline src/File.jsx]: Yo, holmes, if you use a `() =>` function, then you dont need to explicitly `bind` here. Just sayin.

File: src/File.jsx [hunk 2]
```diff
@@ -24,23 +31,36 @@ class File extends React.Component {
...
+
     // when the user clicks a remove link
-    Dispatcher.register('remove-file-preview', function(action, data){
-      if ( action === constants.actions.FILE_UPLOAD_PREVIEW_REMOVE ) {
+    Dispatcher.register(`${this.props.id}-remove-file-preview`, function(action, data) {
```
Comment (mallenscdhhs) [inline src/File.jsx]: Please change the `function` declaration to a fat arrow function

File: src/File.jsx [hunk 3]
```diff
@@ -67,23 +87,25 @@ class File extends React.Component {
       type: 'file',
       value: files
     };
-    var actionName = this.props.fieldValueChangeAction || constants.actions.FIELD_VALUE_CHANGE;
-    Flux.doAction(actionName, event).then(()=> {
+    var actionName = this.props.fieldValueChangeAction || FIELD_VALUE_CHANGE;
```
Comment (mallenscdhhs) [inline src/File.jsx]: maybe use `let` here instead?

File: src/File.jsx [hunk 4]
```diff
@@ -67,23 +87,25 @@ class File extends React.Component {
...
     });
   }
 
   renderPreview() {
     let files = this.state.files;
+    let self = this;
     if(files.length) {
       return (
         <ul className="file-preview-list man pan">
           {files.map(function(file, fileIdx) {
```
Comment (mallenscdhhs) [inline src/File.jsx]: try using a fat arrow here so we can remove the `let self = this;` statement.

File: src/FileListItem.jsx [hunk 5]
```diff
@@ -3,6 +3,8 @@ import React from 'react';
 import constants from './constants';
 import Action from './Action';
 
+let { FILE_UPLOAD_PREVIEW_REMOVE } = constants.actions;
```
Comment (mallenscdhhs) [inline src/FileListItem.jsx]: should get some JSCS warnings about spaces inside curly braces

File: src/File.jsx (hunk 1 above)
Comment (kjschulz) [inline src/File.jsx]: Updated.

File: src/File.jsx (hunk 2 above)
Comment (kjschulz) [inline src/File.jsx]: Updated.

File: src/File.jsx (hunk 3 above)
Comment (kjschulz) [inline src/File.jsx]: Updated.

File: src/File.jsx (hunk 4 above)
Comment (kjschulz) [inline src/File.jsx]: Updated.

File: src/FileListItem.jsx (hunk 5 above)
Comment (kjschulz) [inline src/FileListItem.jsx]: Updated.

**Pull request merged**